src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ProcurementEventService.java

```diff
@@ -1082,4 +1087,175 @@ public void terminateEvent(final Integer procId, final String eventId, final Ter
     updateStatusAndDates(principal, event);
   }
 
+  /**
+   * @param procId
+   * @param eventId
+   * @return
+   */
+  @Transactional
+  public List<SupplierAttachmentResponse> getSupplierAttachmentResponses(
+      final Integer procId, final String eventId) {
+    return getSupplierAttachmentResponses(procId, eventId, null);
+  }
+
+  /**
+   * @param procId
+   * @param eventId
+   * @param supplierId
+   * @return
+   */
+  @Transactional
+  public SupplierAttachmentResponse getSupplierAttachmentResponse(
+      final Integer procId, final String eventId, final String supplierId) {
+
+    var procurementEvent = validationService.validateProjectAndEventIds(procId, eventId);
+    var exportRfxResponse = jaggaerService.getRfx(procurementEvent.getExternalEventId());
+
+    // Determine Jaggaer supplier id
+    var supplierOrganisationMapping =
+        retryableTendersDBDelegate
+            .findOrganisationMappingByOrganisationId(supplierId)
+            .orElseThrow(
+                () ->
+                    new IllegalArgumentException(
+                        String.format(ERR_MSG_FMT_SUPPLIER_NOT_FOUND, supplierId)));
+
+    Optional<Supplier> requestSupplier =
+        exportRfxResponse.getSuppliersList().getSupplier().stream()
+            .filter(
+                supplier ->
+                    supplier
+                        .getCompanyData()
+                        .getId()
+                        .equals(supplierOrganisationMapping.getExternalOrganisationId()))
+            .findFirst();
+
+    List<SupplierAttachmentResponse> supplierAttachmentResponsesList =
+        getSupplierAttachmentResponses(
+            procId, eventId, supplierOrganisationMapping.getExternalOrganisationId().toString());
+
+    return supplierAttachmentResponsesList.stream().findFirst().get();
+  }
+
+  private List<SupplierAttachmentResponse> getSupplierAttachmentResponses(
+      final Integer procId, final String eventId, final String supplierId) {
+
+    var procurementEvent = validationService.validateProjectAndEventIds(procId, eventId);
+    var exportRfxResponse = jaggaerService.getRfx(procurementEvent.getExternalEventId());
+    var supplierList = exportRfxResponse.getSuppliersList();
+
+    if (ObjectUtils.allNotNull(
+        exportRfxResponse.getOffersList(), exportRfxResponse.getOffersList())) {
```

Comment:
Seem to be repeating the argument twice here, did you mean to check `exportRfxResponse.getOffersList().getOffer()` on the second?

Reply:
ah, correct . missed out  while refactoring